### ISSUE TITLE
I would like to get some ideas on how to measure the force and torque between body and body.

### ISSUE THREAD

**Issue by 2-woo-10** [google-deepmind/mujoco#1597]
Hi, I'm a student and I'm trying to use MuJoCo for estimation of the state of holding an object/

I'm looking for some help with how you can measure sensor values

Here is a model which explains my question:

Fingers with the following sensors were modeled.

Currently, the orange part is the force/torque sensor. 
I would like to use mujoco simulator to measure Force and torque in simulator like real environment.

Is there a way to implement this using mujoco simulator? 
I would really appreciate it if you could tell me how.

![image](https://github.com/google-deepmind/mujoco/assets/109202740/cb2b2adf-ee70-43a4-9954-b310934ac0d6)

Here is a screenshot / video, illustrating my question:

**Comment by Balint-H** [comment]
Hello,

If I understand correctly, you want to measure the force/torque between the parent and child bodies? There are sensors to do exactly that! https://mujoco.readthedocs.io/en/stable/XMLreference.html#sensor-force

**Comment by Balint-H** [comment]
Hello, the site should be on the child body, check out the "site" part of the force sensor documentation for details
________________________________

**Comment by 2-woo-10** [comment]
**Hello, Is this what you said correct?

        <body name="Base" pos="0.1 0.1 0.1">
            <geom name="Base"  mass="1"  size="0.05 0.05 0.05" type="box" rgba="1 0 0 0.5"/>
            <body name="Parent_Body" pos="0.0 0.0 0.055">
                <geom name="Parent_Body"  mass="10e-6"  size="0.05 0.05 0.005" type="box" rgba="0 1 0 0.5"/>
                <site name="ft_sensor" type="box" pos="0 0 0.0" />
                <body name="Child_Body" pos="0.0 0.0 0.055">
                    
                    <geom name="Child_Body" type="box" mass="1" size="0.05 0.05 0.05" rgba="0 0 1 0.5"/>        
                </body>
            </body>
        </body>**

**Comment by Balint-H** [comment]
Yeah, except remove the site from the parent body and uncomment it in the child body? Or does that give you an error?

**Comment by 2-woo-10** [comment]
Yeah, The sensor value is still strange
![image](https://github.com/google-deepmind/mujoco/assets/109202740/84d7f661-1715-4a28-b283-e3921ccbdefd)

**Comment by Balint-H** [comment]
![image](https://github.com/google-deepmind/mujoco/assets/41113387/74a2ec3f-041c-4e1d-82b6-bd5211d5b7fd)

Trying your MWE gives correct results as expected. The huge values show on your figure makes me think there's some collision issue. Maybe a plane oriented the wrong way? You can visualize contact points in Simulate, or try disabling collisions in the Physics options.

**Comment by 2-woo-10** [comment]
![image](https://github.com/google-deepmind/mujoco/assets/109202740/c8297fae-59cb-47a6-a19f-c463c6175d07)

Hello, I'm sorry for the late thank you for your response.

It helped me a lot through sincere answers.

I have an additional question, so I'd like to ask you, is it possible?

1. I added the ft sensor, and even if the weight of the object changes while the gripper is closed, the data measured in the ft sensor appears the same.
For example, if a 1kg object holds and lifts an object when the joint is free, and if there is no joint and the object does not move, it will slip. I want to know why the data appear the same.

**Comment by Balint-H** [comment]
Glad you are making progress. Could you please rephrase your point 1? I had trouble understanding it.

Could you please share the xml, or describe the hierarchy and where the sensor is?

**Comment by 2-woo-10** [comment]
![image](https://github.com/google-deepmind/mujoco/assets/109202740/ffcba899-56fc-490c-946b-36a4e9f7f5b8)

Currently, a force/Torque sensor has been attached to that part. 

The experiment I want to proceed with is that when I hold the object with the gripper and the gripper moves up and down, I want to compare the ft sensor data.

Case 1 :

    <body pos="0.13 0 0.08">
      <joint type="free"/>
      <geom name="object" type="box" mass="5" size="0.015 0.015 0.08" rgba="0.1 0 0.5 0.1"/>
    </body> 

The object was declared as follows, and the data were obtained when the gripper held the object stably and moved upward.

Case 2 : 

    <body pos="0.13 0 0.08">
      <geom name="object" type="box" mass="5" size="0.015 0.015 0.08" rgba="0.1 0 0.5 0.1"/>
    </body> 

This time, the object is fixed, and I got the data that appears when the gripper holds and moves the object. (In this case, the object does not move and only the gripper moves -> Slip)

In this case, the data appears the same, and I wonder if it is measured correctly..

Additionally, is it correct that the data appear similar even if the experiment is conducted in the same way by changing the mass of the object?
For example, when there are 1kg and 5kg objects, holding the gripper with the same grasping force and moving the gripper upward, the data on the f/t sensor is the same.

**Comment by Balint-H** [comment]
Well, if the object is fixed, you expect the same data measured with any mass (as the slip conditions would stay the same). Do you still see the same measurements when the object is free as well and lifted?

**Comment by 2-woo-10** [comment]
Yes.
In the case of a free object, it also shows the same measurement values.
